.github/actions/test_gem/action.yml

```diff
@@ -0,0 +1,130 @@
+      shell: bash
+      run: |
+        dir=$(find . -iname '${{ inputs.gem }}.gemspec' -exec dirname {} \;)
+        echo "::set-output name=dir::${dir}"
+
+        # We install multiple ruby versions here, and that makes for some
+        # annoying bundler conflicts when we get to the JRuby step. Removing
+        # the lockfile slows things down a bit, but we should still get most
+        # of the benefits of bundler caching.
+        rm -f "${dir}/Gemfile.lock"
+
+        echo "::set-output name=cache_key::mri"
+        if [[ "${{ inputs.ruby }}" == "jruby" ]]; then
+          echo "::set-output name=cache_key::jruby"
+        fi
+
+        echo "::set-output name=appraisals::false"
+        if [[ -f "${dir}/Appraisals" ]]; then
+          echo "::set-output name=appraisals::true"
+        fi
+
+    # Install ruby and bundle dependencies and cache!
+    # ...but not for appraisals, sadly.
+    - name: Install Ruby ${{ inputs.ruby }} with dependencies
+      if: "${{ steps.setup.outputs.appraisals == 'false' }}"
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: "${{ inputs.ruby }}"
+        working-directory: "${{ steps.setup.outputs.dir }}"
+        bundler-cache: true
+        cache-version: "v1-${{ steps.setup.outputs.cache_key }}"
+
+    # If we're using appraisals, do it all manually.
+    - name: Install Ruby ${{ inputs.ruby }} without dependencies
+      if: "${{ steps.setup.outputs.appraisals == 'true' }}"
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: "${{ inputs.ruby }}"
+    - name: Install dependencies and appraisals
+      if: "${{ steps.setup.outputs.appraisals == 'true' }}"
+      shell: bash
+      run: |
+        cd "${{ steps.setup.outputs.dir }}"
+        bundle install
+        bundle exec appraisal install
+
+    - name: Test Gem
+      shell: bash
+      run: |
+        cd "${{ steps.setup.outputs.dir }}"
+
+        if [[ -f "Appraisals" ]]; then
+          bundle exec appraisal rake test
+        else
+          bundle exec rake test
+        fi
+      env:
+        TEST_KAFKA_HOST: "127.0.0.1"
+        TEST_KAFKA_PORT: 29092
+        TEST_MYSQL_HOST: "127.0.0.1"
+        TEST_MYSQL_PORT: 3306
+        TEST_MYSQL_USER: mysql
+        TEST_MYSQL_PASSWORD: mysql
+        TEST_POSTGRES_PASSWORD: postgres
+        TEST_POSTGRES_USER: postgres
+        TEST_POSTGRES_HOST: localhost
+        TEST_POSTGRES_PORT: 5432
+        TEST_POSTGRES_DB: postgres
+        TEST_MEMCACHED_HOST: localhost
+        TEST_MEMCACHED_PORT: 11211
+        TEST_MONGODB_HOST: localhost
+        TEST_MONGODB_PORT: 27017
+        TEST_RABBITMQ_HOST: localhost
+        TEST_RABBITMQ_PORT: 5672
+        TEST_RABBITMQ_URL: amqp://guest:guest@localhost:5672
+        TEST_REDIS_HOST: localhost
+        TEST_REDIS_PORT: 6379
+
+    - name: YARD
+      shell: bash
+      if: "${{ inputs.yard == 'true' }}"
+      run: |
+        cd "${{ steps.setup.outputs.dir }}"
+        bundle exec rake yard
+
+    - name: Rubocop
```

Comment:
I do not think we need to run this for every gem version. 

Would it be OK to run Rubocop only for the minimal version of Ruby we intend on supporting?

Reply:
> Would it be OK to run Rubocop only for the minimal version of Ruby we intend on supporting?

Yes, and

> I do not think we need to run this for every gem version.

we aren't! One of the downsides of this change is that it doesn't make our CI system _much_ more clear - although I believe it helps clarify that to a small degree.

The relevant bits of code are [here, defining a default 'rubocop' input to 'false'](https://github.com/open-telemetry/opentelemetry-ruby/pull/1249/files#diff-e018a7f25c2bd16ae50ee7acbd551cb6a4afa64811f45451b454e0896e0d6e0cR17-R21); [here, where we only run the rubocop step if the variable is in fact 'true'](https://github.com/open-telemetry/opentelemetry-ruby/pull/1249/files#diff-e018a7f25c2bd16ae50ee7acbd551cb6a4afa64811f45451b454e0896e0d6e0cR120); and finally [here, where we only set that variable to true for Ruby 2.7 builds](https://github.com/open-telemetry/opentelemetry-ruby/pull/1249/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR45-R53) (and that last pattern is repeated for every build group unless I messed it up somehow).

So the net effect is "we are only running Rubocop (and the YARD / `gem build` smokescreen tests) for Ruby 2.7".

Reply:
Thank you for explaining that!!! Do I even work at GH?

Reply:
🤣